File: issta2018artifacts/minimized-bugs/buggy-key/GetSimplifyFreeIndices.java
// Simplified test case for Daikon commit 29dfd8dde

public class GetSimplifyFreeIndices {

	/*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
	public static Object get_simplify_free_indices(Object... vars) {
	if (vars.length == 1) {
	    return vars[0];
	} else if (vars.length == 2) {
	    //:: error: array.access.unsafe.high.constant
	    return compose(vars[0], vars[2]);
	} else {
		return null;
	}
    }

	/*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
    public static Object compose(Object obj1, Object obj2) {
	return null;
    }
}

File: issta2018artifacts/minimized-bugs/buggy-clousot/Daikon/GetSimplifyFreeIndices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

// Simplified test case for Daikon commit 29dfd8dde

class GetSimplifyFreeIndices
{
    public static object get_simplify_free_indices(params object[] vars)
    {
        if (vars.Length == 1) {
            return vars[0];
        }
        else if(vars.Length == 2)
        {
            //:: error: array.access.unsafe.high.constant
            return compose(vars[0], vars[2]);
        }
        else
        {
            throw new Exception("unexpected length" + vars.Length);
        }
    }
    public static object compose(object ob1, object obj2)
    {
        return null;
    }
}
